src/selections/lexer.cpp

```diff
@@ -195,14 +195,14 @@ Token Tokenizer::variable() {
             break;
         }
     }
-    int data = 0;
+    size_t data = 0;
     auto number = input_.substr(start, count);
     try {
-        data = std::stoi(number);
+        data = parse<size_t>(number);
     } catch (const std::exception&) {
         throw selection_error("could not parse number in '{}'", number);
     }
-    if (data > UINT8_MAX) {
+    if (data > static_cast<size_t>(UINT8_MAX)) {
```

Comment:
Would `std::numeric_limits` be better here?

Reply:
According to https://en.cppreference.com/w/cpp/types/numeric_limits, there is no specialization of `std::numeric_limits ` for `uint8_t` in the standard. So UINT8_MAX might be better.